presto-spi/src/main/java/com/facebook/presto/spi/translator/TargetExpression.java

```diff
@@ -0,0 +1,60 @@
+package com.facebook.presto.spi.translator;
+
+import com.facebook.presto.spi.relation.RowExpression;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+public class TargetExpression<T>
+{
+    private RowExpression originalExpression;
+    private List<TargetExpression> params;
+    private Optional<T> translatedExpression;
+
+    public TargetExpression(Optional<T> translatedExpression)
```

Comment:
Rough implementation of TargetExpression, the fields are accurate but most of these methods will be changed.